### PR TITLE
HW 8: Task 1

### PR DESCRIPTION
**Pull request by ValeriL**

### CODE REVIEW

File: Homework/homework8/task1.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+(for example when there's a line 1=something) ValueError should be raised.
+File size is expected to be small, you are permitted to read it entirely into memory.
+"""
+
+
+class KeyValueStorage:
+    def __init__(self, path):  # noqa : CCR001
+        with open(path) as file:
+            for line in file:
+                attr, value = line.replace("\n", "").split("=")
```
Comment (Cjkjvfnby) [inline Homework/homework8/task1.py:32]: `line.replace("\n", "")`  check st methods, there is a better way to do it.

File: Homework/homework8/task1.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                if attr.isdigit():
+                    raise ValueError("Int cant be an attribute")
+                if value.isdigit():
+                    self.__dict__[attr] = int(value)
+                else:
+                    self.__dict__[attr] = value
```
Comment (Cjkjvfnby) [inline Homework/homework8/task1.py:38]: Please check requirements.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: Homework/homework8/task1.py (hunk 2 above)
Comment (ValeriL) [reply Homework/homework8/task1.py:38]: But whats wrong?
> If a value can be treated both as a number and a string, it is treated as number.
Or is mistake in different?

Comment (Cjkjvfnby) [reply Homework/homework8/task1.py:38]: It is about another requirement.  Just add tests for all cases.

File: Homework/homework8/test_task1.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    with pytest.raises(ValueError, match="Int cant be an attribute"):
+        KeyValueStorage(temp_file)
+
+
+@pytest.mark.parametrize("text", ["power=9001\n"])
+def test_treated_as_int(temp_file):
+    assert isinstance(KeyValueStorage(temp_file).power, int)
+
+
+@pytest.mark.parametrize("text", ["name=name1\nname=name2\n"])
```
Comment (Cjkjvfnby) [inline Homework/homework8/test_task1.py:35]: `__str__=foo`

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**